blogposts/2021/api-documentation-generated-for-all-your-code.md

```diff
@@ -0,0 +1,84 @@
+
+![API Docs button](https://sourcegraphstatic.com/blog/api-docs/api-docs-button.png)
+
+
+Once you click it, you'll be greeted with information about how to get started in a brand new welcome prompt we’ve added:
+
+![Welcome prompt](https://sourcegraphstatic.com/blog/api-docs/welcome-prompt.png)
+
+### Better sub-page navigation
+
+Within a Go package, there are many sub-packages. We've improved the way we render documentation subpages so you can skip over useless directories and get right to the content (but we're still hard at work improving navigation in API docs!):
+
+![Sub-page navigation](https://sourcegraphstatic.com/blog/api-docs/subpage-navigation.gif)
+
+### Understands the types of symbols, and uses iconography to communicate
+
+Due to API docs being built on code intelligence data, it understands exactly what symbols are and has detailed information about them. Whether symbols are deprecated, whether they're a function, variable, interface, etc. and the UI now communicates this information using icons:
+
+![Symbols information](https://sourcegraphstatic.com/blog/api-docs/symbols.gif)
+
+### A deep understanding of code
+
+We're paying attention to the small details, and really tailoring API docs to each language. We want an experience that makes sense for Go first, because here at Sourcegraph we use primarily Go and TypeScript and want to ensure we build an experience Go developers love. Then we'll expand to more languages and build an experience that makes sense in the context of other languages—a tailored experience for each.
+
+One way we've enabled this is by allowing LSIF language indexers to emit data in the way they see fit. Sourcegraph merely renders it, indexes it, and adds a nice navigation experience. For example, the Go LSIF indexer even chooses to display URL hashes in a way that makes sense specifically for Go, with a `Receiver.Method` format:
+
+![URL hash display for Go](https://sourcegraphstatic.com/blog/api-docs/url-hash-displays.gif)
+
+In the near future, we'll enable you to search and filter based on symbol types as well: "just show me variables", "just private and deprecated functions", etc. based on the deep understanding of the code we've built.
+
+![Display symbol types](https://sourcegraphstatic.com/blog/api-docs/symbol-types.gif)
+
+### Feedback widget
+
+We know there will be a ton of feedback as things are very early in development, and so we've made it as easy as possible to give feedback from any API docs page:
+
+![Feedback widget](https://sourcegraphstatic.com/blog/api-docs/feedback-widget.png)
+
+You can also directly email feedback to the lead engineer [stephen@sourcegraph.com](mailto:stephen@sourcegraph.com) (that's me) or the entire team via Twitter [@sourcegraph](https://twitter.com/sourcegraph). We really want to hear what you think!
```

Comment:
@slimsag Do you want to add an author bio for yourself (example [here](https://about.sourcegraph.com/blog/how-not-to-break-a-search-engine-unglamorous-engineering/))? It's optional so up to you!

Reply:
Done!